modules/structs/commit_status.go

```diff
@@ -16,13 +16,16 @@ const (
 	CommitStatusError CommitStatusState = "error"
 	// CommitStatusFailure is for when the CommitStatus is Failure
 	CommitStatusFailure CommitStatusState = "failure"
+	// CommitStatusWarning is for when the CommitStatus is Warning
+	CommitStatusWarning CommitStatusState = "warning"
 )
 
 var commitStatusPriorities = map[CommitStatusState]int{
 	CommitStatusError:   0,
 	CommitStatusFailure: 1,
 	CommitStatusPending: 2,
 	CommitStatusSuccess: 3,
+	CommitStatusWarning: 4,
```

Comment:
I think it is supposed to be
```suggestion
	CommitStatusWarning: 2,
	CommitStatusPending: 3,
	CommitStatusSuccess: 4,
```
(assuming we want to say that a `warning` state is worse than a still pending status as the pending status might also be successful)

Reply:
Yep, I just double-checked, that's also the previous behavior.

Reply:
Thanks, I was getting very confused at that old code. Let's see if tests pass with this change.